cmd/otelcol/flags.go

```diff
@@ -0,0 +1,100 @@
+// Copyright  Splunk, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"flag"
+	"io"
+	"strings"
+
+	"go.opentelemetry.io/collector/service/featuregate"
+)
+
+var (
```

Comment:
What do you think about putting these variables into a struct and returning that from a function that takes the command line args? Like this:
```
func main() {
	flags, err := parseFlags(os.Args[1:])
	// ...
}

type flags struct {
	a bool
	b int
	c string
}

func parseFlags(args []string) (flags, error) {
	flagSet := new(flag.FlagSet)
	out := flags{}
	flagSet.BoolVar(&out.a, "a", false, "usage")
	flagSet.IntVar(&out.b, "b", 42, "usage")
	flagSet.StringVar(&out.c, "c", "default", "usage")
	err := flagSet.Parse(args)
	return out, err
}
```
Then we can avoid using package (global) variables and we have a testable, self-contained function.

Reply:
Awesome. Yeah I think the Flag suffixes can go away now since the struct is called flags.

Reply:
I'll remove the `Flag` suffixes, thanks!